### PR TITLE
Breaking: #96287 - Doctrine DBAL 3.2

### PR DESCRIPTION
**Pull request by linawolf**
references https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/1624
solves https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/1593

make another, slightly changed PR for 11.5

releases: master

### CODE REVIEW

**Comment by linawolf** [comment]
Do not merge without an ok from @lolli42

File: Documentation/Testing/ExtensionTesting.rst
Comment (sbuerk) [inline Documentation/Testing/ExtensionTesting.rst:558]: We should add a (int) cast here. It may be not an integer which is returned (for ex. with sqlite and PHP before 8.1)

File: Documentation/Testing/ExtensionTesting.rst
Comment (sbuerk) [inline Documentation/Testing/ExtensionTesting.rst:570]: Same here, (int) cast

**Review by sbuerk — CHANGES_REQUESTED** [review]
Thanks for your work Lina, but I have minor remarks which should be addressed.

File: Documentation/ApiOverview/Database/Statement/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/Statement/Index.rst:8]: We will update the prepared statement section lateron. This chapter never described more then how to get the results. The change doesn't hinder us from explaining statements in DBAL context lateron

File: Documentation/ApiOverview/Workspaces/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Workspaces/Index.rst:260]: I added a todo, I think we cannot solve this from the scope of this issue.

File: Documentation/ApiOverview/Database/Statement/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/Statement/Index.rst]: Updated it

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:341]: could you make another PR about this one?

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:751]: I rephrased this to make the relation to Doctrine DBAL 3.2 more clear. Doctrine DBAL 3.2 has been introduced with 12 not with 11. We can still put another message hinting to the migration plan in v11 here.

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:809]: The double point combined with an indentation creates a code-block, it is a short-command... 

Should we put a todo that the example needs to be rebuild?

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (sbuerk) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:751]: Yes, but as said, in the core provided QueryBuilder we have added these two methods on our own, and also backported to v11. That's working independent from doctrine/dbal version.

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (sbuerk) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:809]: Either the todo - or take out at least the "and JOIN expression".

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (sbuerk) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:341]: At a later point, yes.

File: Documentation/ApiOverview/Database/Statement/Index.rst
Comment (sbuerk) [reply Documentation/ApiOverview/Database/Statement/Index.rst:8]: Ack.

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:751]: So what kind of formulation shall we use here?

**Comment by linawolf** [comment]
@sbuerk what else is needed to get this merged?

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (sbuerk) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst:751]: .. versionchanged:: 11.5.4

File: Documentation/Testing/ExtensionTesting.rst
Comment (sbuerk) [inline Documentation/Testing/ExtensionTesting.rst:558]: without space between cast and variable

File: Documentation/Testing/ExtensionTesting.rst
Comment (sbuerk) [inline Documentation/Testing/ExtensionTesting.rst:570]: without space between cast and variable

File: Documentation/ApiOverview/Workspaces/Index.rst
Comment (sbuerk) [inline Documentation/ApiOverview/Workspaces/Index.rst:145]: empty line superflous (one empty line too much)

**Review by sbuerk — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Workspaces/Index.rst
Comment (sbuerk) [reply Documentation/ApiOverview/Workspaces/Index.rst:145]: @linawolf otherwise, beside these three small things, it's ok from my side.

**Comment by sbuerk** [comment]
> @sbuerk what else is needed to get this merged?

@linawolf otherwise, beside the three small things in the review above, it's ok from my side.

File: Documentation/ApiOverview/Database/QueryBuilder/Index.rst
Comment (linawolf) [reply Documentation/ApiOverview/Database/QueryBuilder/Index.rst]: we only keep two versions, but changed it to 11.5

**Review by sbuerk — APPROVED** [review]

**Review by lolli42 — APPROVED** [review]

**Pull request merged**
